pyomo/contrib/mindtpy/util.py

```diff
@@ -181,3 +181,60 @@ def var_bound_add(solve_data, config):
                         var.setub(config.integer_var_bound)
                     else:
                         var.setub(config.continuous_var_bound)
+
+
+def generate_L2_objective_function(model, setpoint_model, discretes_only=False):
+    """Generate objective for minimum euclidean distance to setpoint_model
+    L2 distance of (x,y) = \sqrt{\sum_i (x_i - y_i)^2}
+    discretes_only -- only optimize on distance between the discrete variables
+    """
+    var_filter = (lambda v: v[1].is_binary()) if discretes_only \
+        else (lambda v: True)
+
+    model_vars, setpoint_vars = zip(*filter(var_filter,
+                                            zip(model.component_data_objects(Var),
+                                                setpoint_model.component_data_objects(Var))))
+    assert len(model_vars) == len(
+        setpoint_vars), "Trying to generate L1 objective function for models with different number of variables"
+
+    return Objective(expr=(
+        sum([(model_var - setpoint_var.value)**2
+             for (model_var, setpoint_var) in
+             zip(model_vars, setpoint_vars)])))
```

Comment:
Here you are never computing the squared root. I would change the function documentation and even the function name to `generate_Norm2sq_objective_function`